components/frontend/src/header_footer/buttons/DownloadAsPdfButton.test.jsx

```diff
@@ -6,15 +6,14 @@ import * as fetchServerApi from "../../api/fetch_server_api"
 import * as toast from "../../widgets/toast"
 import { DownloadAsPdfButton } from "./DownloadAsPdfButton"
 
-vi.mock("../../api/fetch_server_api.js")
-vi.mock("../../widgets/toast.jsx")
-
 beforeEach(() => {
-    vi.resetAllMocks()
     history.push("")
-    fetchServerApi.fetchServerApi.mockImplementation(() => Promise.resolve({ ok: true }))
+    vi.spyOn(fetchServerApi, "fetchServerApi").mockImplementation(() => Promise.resolve({ ok: true }))
+    vi.spyOn(toast, "showMessage")
 })
 
+afterEach(() => vi.resetAllMocks())
```

Comment:
I don't understand when `afterEach(() => vi.resetAllMocks())` is used; it seems unneeded when a `spyOn` mock promise is placed in a `beforeAll`, but only occasionally used when `beforeEach` contains a `spyOn` mock promise?

Reply:
You're right. No `resetAllMocks` is necessary when using `spyOn`. Only when using `vi.mock(module)` it is needed.